source/agora/consensus/protocol/Nominator.d

```diff
@@ -1064,8 +1059,44 @@ extern(D):
         () @trusted { this.seen_envs.clear(); }();
     }
 
+    /// If we have majority signatures then externalize to the ledger otherwise check again after receiving signatures
+    extern(D) private void checkExternalize () @safe nothrow
+    {
+        const block = this.pending_block;
+        try
+        {
+            if (block.header.height == this.ledger.height + 1)
+            {
+                if (this.ledger.hasMajoritySignature(block.header))
```

Comment:
At this point we should just remove `hasMajoritySignature` and check `percentage`.

Reply:
In `Ledger.hasMajoritySignature` it does exactly that. Or do you want to avoid the function call?

Reply:
Well it's not a property of the `Ledger` anymore: We use it on non-externalize blocks, and it doesn't depend on `ConsensusParams`, so I don't know why it should stay in the `Ledger`.

Reply:
Updated